queries/rust/highlights.scm

```diff
@@ -99,6 +99,8 @@
 
 (function_item (identifier) @function)
 (function_signature_item (identifier) @function)
+((impl_item (declaration_list (function_item (identifier) @constructor)))
+ (#eq? @constructor "new"))
```

Comment:
i'm not sure about that, `new` is a convention, not a language specific thing in rust

Reply:
I agree with that not, it is not even a real convention, but more of a "logic" way to name your contructor.

Reply:
It's a convention, same as naming `self` the first argument in python, we highlight self in python too.

Reply:
I am really not sure, it is not listed in the clippy things.

Reply:
yup and i've seen repos not using new to construct structures sometimes (depends on the use case actually). Self is a convention by itself because it represent the instance of an object in a method, just like in every other language :)

Reply:
I would also rather have this as a regular function

Reply:
> I am really not sure, it is not listed in the clippy things.

Because you can't infer when you are defining a constructor. `new` is all over the standard library and the rust book.